### ISSUE TITLE
Implicit Intent: localIntent used to create instance of PendingIntent (Part of Penetration testing)

### ISSUE THREAD

**Issue by Naveentp** [AltBeacon/android-beacon-library#434]
### Expected behavior
LocalIntents not to be used to create instance of PendingIntent as per OWASP guidelines.

### Actual behaviour
When the app is given for Penetration testing, this issue arised in PendingIntent issues section. The description said,

Implicit Intent: localIntent used to create instance of PendingIntent. A malicious application could potentially intercept, redirect and/or modify (in a limited manner) this Intent. Pending Intents retain the UID of your application and all related permissions, allowing another application to act as yours. File: /media/sf_share/T/app-release/classes_dex2jar/org/altbeacon/beacon/service/BeaconService.java

### Android Beacon Library version
v2+

**Comment by davidgyoung** [comment]
Any specifics on the line number in the BeaconService where this issue was detected?  The only use of `PendingIntent` I see is this snippet, and it doesn't look to me like there is a security issue with it:

```
       Intent restartIntent = new Intent();
        restartIntent.setClassName(getApplicationContext(), StartupBroadcastReceiver.class.getName());
        return getBroadcast(getApplicationContext(), 1, restartIntent, FLAG_ONE_SHOT);
```

That said, if tools are flagging this code (either rightfully or wrongfully) it may be good to rewrite it so this is not an issue.

**Comment by cupakromer** [comment]
@Naveentp do you know the name of the tool which flagged this?

Based on the Android [_Intents and Intent Filters Guide_](https://developer.android.com/guide/components/intents-filters.html) the different between an [explicit intent](https://developer.android.com/guide/components/intents-filters.html#ExampleExplicit) and an [implicit intent](https://developer.android.com/guide/components/intents-filters.html#ExampleSend) is minor:

```java
// Explicit
Intent explicitIntent = new Intent(this, DownloadService.class);
explicitIntent.putExtra(Intent.EXTRA_TEXT, textMessage);

// Implicit
Intent implicitIntent = new Intent();
implicitIntent.putExtra(Intent.EXTRA_TEXT, textMessage);
```

I think the issue may simply be whatever tool is used is statically analyzing the code and is simply flagging `new Intent()`. If we can get the name of the tool we can test that simply updating the calls to be `new Intent(getApplicationContext(), StartupBroadcastReceiver.class)` should resolve it.

**Comment by Naveentp** [comment]
@cupakromer The tool is used for the penetration testing is [Qark](https://github.com/linkedin/qark) and Yes, its a Static code analyzer tool. Implementing explicit intent should resolve this issue.

**Comment by davidgyoung** [comment]
Fixed in #437
